docs/architecture/security-and-data-sovereignty.md

```diff
@@ -0,0 +1,124 @@
+- **Dynamic Credential Provisioning**: Credentials are dynamically generated upon consumer data transfer requests.
+- **Access Control Lists (ACLs)**: OAuth scopes are mapped to Kafka ACLs to grant precise topic access.
+- **Token-Based Authorization**: OAuth tokens control Kafka topic access with clearly defined scopes and expiration times.
+
+### 1.2 Access Revocation
+
+The procedure for revoking consumer access includes:
+
+1. **Suspend/Terminate Transfer**: Provider initiates suspension or termination via Control Plane.
+2. **Token Revocation**: Kafka Extension instructs Data Plane and OAuth Service to revoke tokens.
+3. **Access Termination**: OAuth Service confirmation ceases consumer access.
+
+### 1.3 Component Responsibilities
+
+Explicit responsibilities for data flow control:
+
+- **Control Plane**: Manages negotiations, policy compliance, and interaction with Kafka Extension.
+- **Kafka Extension**: Coordinates credential management between Control and Data Planes.
+- **OAuth Service**: Handles dynamic client registrations, token lifecycle management.
+- **Vault**: Securely stores authentication credentials and secrets.
+
+## 2. Token Management
+
+### 2.1 Token Roles
+
+Tokens utilized by the Kafka extension:
+
+- **Access Tokens**: Short-lived JWT tokens facilitating immediate Kafka topic access.
+
+### 2.2 Token Lifecycle
+
+Scenarios managed by token lifecycle processes:
+
+- **Normal Expiration**: Automated token refresh prior to expiration.
+- **Forced Revocation**: Immediate token revocation capability.
+- **Failed Refresh**: Consumer access terminated upon token refresh failure.
+
+### 2.3 Asset Role
+
+- The **Kafka topic** is the negotiated asset, not individual records.
+- Topic-level negotiation and access control.
+- Consumers access all messages published during token validity.
+
+## 3. Security Measures
+
+### 3.1 Security Protocols
+
+Kafka Extension security protocols include:
+
+- **Transport Encryption**: Secure communications using TLS/SSL (SASL_SSL protocol).
+- **Authentication**: OAuth 2.0 tokens via SASL/OAUTHBEARER.
+- **Authorization**: Comprehensive authorization via EDC policies, Kafka ACLs, and OAuth scopes.
+
+### 3.2 Protection Against Unauthorized Access
+
+Security features preventing unauthorized access:
+
+- **Dynamic Credentials**: Temporary, single-use credentials.
+- **Short-lived Tokens**: Reducing potential misuse windows.
+- **Secure Storage**: Credentials managed securely in Vault.
+- **Token Validation**: OAuth Service token validation prior to Kafka access.
+
+## 4. Self-Sovereign Data Sharing
+
+### 4.1 Autonomous Control
+
+Providers maintain full autonomous control over data sharing:
+
+- **Policy Definition**: Customizable data access policies.
+- **Dynamic Management**: Real-time policy enforcement via Control Plane.
+- **Data Residency**: Data remains on provider infrastructure, access-only model.
+- **Contractual Governance**: Data access governed by explicitly negotiated contracts.
```

Comment:
The notion of “full autonomous control” appears to contradict “contractual governance,” which implies external constraints.

Recommendation:
Clarify that contracts are instruments of self-determination, not limitations. They enable autonomous actors to define the terms of their interactions rather than restricting their autonomy.